src/go/k8s/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -183,6 +183,37 @@ type ClusterSpec struct {
 
 	// The name of the ServiceAccount to be used by the Redpanda pods
 	ServiceAccount *string `json:"serviceAccount,omitempty"`
+
+	// NodePools runs a StatefulSet for every NodePool. All these produce pods,
+	// that join the cluster, but are steered independently (especially replica
+	// count).
+	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
+}
+
+// NodePoolSpec defines a NodePool. NodePools have their own:
+// NodeSelector, so they can be scheduled on specific cloud provider Node Pools.
+// Storage, as different NodePools may have different disk shapes.
+// Replicas, so they can be controlled independently
+// Resources, because this is tied strongly to the actual machine shape backing the NodePool.
+type NodePoolSpec struct {
+	Name string `json:"name"`
+	// Replicas determine how big the node pool will be.
+	// +kubebuilder:validation:Minimum=0
+	Replicas *int32 `json:"replicas,omitempty"`
+	// If specified, Redpanda Pod tolerations
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// If specified, Redpanda Pod node selectors. For reference please visit
+	// https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Storage spec for cluster
+	Storage StorageSpec `json:"storage,omitempty"`
```

Comment:
Should this be `*StorageSpec` or not have `omitempty`?

Reply:
this is copied from the pre-existing spec. there, it is like this:
```
	Storage StorageSpec `json:"storage,omitempty"`
```

all structs have omityempty. not sure what the purpose of omityempty on non-pointer structs is tbh, will it be omitted from json marshal if == empty struct? i suppose this has pretty much no effect..

but now that i see it:

it looks like we should:

- change spec.storage to being pointer-to-struct and thus optional, so i can (as with other fields) create a Cluster without a default nodePool, but only spec.nodePools.

Reply:
removed the omitempty and added +required annotation